utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel

```diff
@@ -0,0 +1,169 @@
+    deps = [
+        ":annotations",
+        ":log_level",
+        "//third_party:javax_inject_javax_inject",
+        "//third_party:org_jetbrains_kotlinx_kotlinx-coroutines-core",
+        "//utility/src/main/java/org/oppia/android/util/threading:annotations",
+    ],
+)
+
+kt_android_library(
+    name = "event_bundle_creator",
+    srcs = [
+        "EventBundleCreator.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        "//model:event_logger_java_proto_lite",
+    ],
+)
+
+kt_android_library(
+    name = "event_logger",
+    srcs = [
+        "EventLogger.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        "//model:event_logger_java_proto_lite",
+    ],
+)
+
+kt_android_library(
+    name = "exception_logger",
+    srcs = [
+        "ExceptionLogger.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+)
+
+kt_android_library(
+    name = "firebase_event_logger",
+    srcs = [
+        "firebase/FirebaseEventLogger.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":event_bundle_creator",
+        ":event_logger",
+        "//model:event_logger_java_proto_lite",
+        "//third_party:com_google_firebase_firebase-analytics",
+        "//third_party:javax_inject_javax_inject",
+    ],
+)
+
+kt_android_library(
+    name = "firebase_exception_logger",
+    srcs = [
+        "firebase/FirebaseExceptionLogger.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":exception_logger",
+        "//third_party:com_google_firebase_firebase-crashlytics",
+    ],
+)
+
+kt_android_library(
+    name = "firebase_log_reporting_module",
+    srcs = [
+        "firebase/LogReportingModule.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":dagger",
+        ":event_bundle_creator",
+        ":firebase_event_logger",
+        ":firebase_exception_logger",
+        "//third_party:com_google_firebase_firebase-analytics",
+        "//third_party:com_google_firebase_firebase-crashlytics",
+        "//third_party:javax_inject_javax_inject",
+    ],
+)
+
+kt_android_library(
+    name = "firebase_log_uploader",
+    srcs = [
+        "firebase/FirebaseLogUploader.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":log_uploader",
+        "//third_party:androidx_work_work-runtime",
+        "//third_party:androidx_work_work-runtime-ktx",
+        "//third_party:javax_inject_javax_inject",
+    ],
+)
+
+kt_android_library(
+    name = "firebase_log_uploader_module",
+    srcs = [
+        "firebase/FirebaseLogUploaderModule.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":dagger",
+        ":firebase_log_uploader",
+        ":log_uploader",
+    ],
+)
+
+kt_android_library(
+    name = "log_level",
+    srcs = [
+        "LogLevel.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
```

Comment:
Is it possible to omit the visibility field and make this library private? It seems like it is only used within the package itself.

Reply:
Ok I'll do it private

Reply:
Done